src/color_laws.jl

```diff
@@ -452,3 +452,110 @@ function f04_invum(x::Real, Rv::Real)
+    - 0.77530 * (m14_x2_diff)^6
+    + 0.32999 * (m14_x2_diff)^7
+    + (0.0, 0.0, -0.011, 0.0, 0.0)
+)
+const m14_b2v = @. (
+      1.41338 * (m14_x2 - 1.82)
+    + 2.28305 * (m14_x2 - 1.82)^2
+    + 1.07233 * (m14_x2 - 1.82)^3
+    - 5.38434 * (m14_x2 - 1.82)^4
+    - 0.62251 * (m14_x2 - 1.82)^5
+    + 5.30260 * (m14_x2 - 1.82)^6
+    - 2.09002 * (m14_x2 - 1.82)^7
+    + (0.0, 0.0, 0.091, 0.0, 0.0)
+)
+const m14_a3v = @. (
+      1.752
+    - 0.316 * m14_x3
+    - 0.104 / ((m14_x3 - 4.67)^2 + 0.341)
+    + (0.442, 0.341, 0.130, 0.020, 0.000)
+)
+#const m14_a3d = -0.316 + 0.104 * 2.0 * (m14_xi3 - 4.67) / ((m14_xi3 - 4.67)^2 + 0.341)^2
+const m14_b3v = @. (
+     -3.090
+    + 1.825 * m14_x3
+    + 1.206 / ((m14_x3 - 4.62)^2 + 0.263)
+    - (1.256, 1.021, 0.416, 0.064, 0.000)
+)
+#const m14_b3d = 1.825 - 1.206 * 2 * (m14_xi3 - 4.62) / ((m14_xi3 - 4.62)^2 + 0.263)^2
+const m14_xn = collect((m14_x1..., m14_x2..., m14_x3...))
+const m14_anv = collect((m14_a1v..., m14_a2v..., m14_a3v...))
+const m14_bnv = collect((m14_b1v..., m14_b2v..., m14_b3v...))
+const m14_a_spl = Spline1D(m14_xn, m14_anv, bc="nearest")
+const m14_b_spl = Spline1D(m14_xn, m14_bnv, bc="nearest")
+
+"""
+    M14(;Rv=3.1)
+
+Maiz Apellaniz et al (2014) Milky Way & LMC R(V) dependent model.
+
+Returns E(B-V) in magnitudes at the given wavelength relative to the
+extinction. The published UV extinction curve is identical to Clayton,
+Cardelli, and Mathis (1989, CCM). Forcing the optical section to match smoothly
+with CCM introduces a non-physical feature at high values of R5495 around 3.9
+inverse microns; see section 5 in Maiz Apellaniz et al. (2014) for more
+discussion. For that reason, we provide the M14 model only through 3.3 inverse
+microns, the limit of the optical in CCM. Outside of that range this will
+return 0. Rv is the selective extinction and is valid over [2, 6]. A typical
+value for the Milky Way is 3.1. R5495 = A(5485)/E(4405-5495) Spectral
+equivalent to photometric R(V).
+
+# References
+[Maiz Apellaniz et al. (2014)](https://ui.adsabs.harvard.edu/abs/2014A%26A...564A..63M/)
+"""
+@with_kw struct M14 <: ExtinctionLaw
+    Rv::Float64 = 3.1
+end
+
+function (law::M14)(wave::T) where T
+    checkbounds(law, wave) || return zero(float(T))
+    x = aa_to_invum(wave)
+    return m14_invum(x, law.Rv)
+end
+
+bounds(::Type{M14}) = (3030.3, 33333.3)
+
+"""
+    DustExtinction.m14_invum(x, Rv)
+
+The algorithm used for the [`M14`](@ref) extinction law, given inverse microns and Rv. For more information, seek the original paper.
+"""
+function m14_invum(x::Real, Rv::Real)
+    if !(0.3 <= x <= 3.3)
+        error("out of bounds of M14, support is over $(bounds(M14)) angstrom")
+    end
+
+    # Infrared
+    ai = 0.574 * x^1.61
+    bi = -0.527 * x^1.61
+
+    # Optical
+    m14_av = m14_a_spl(x)
+    m14_bv = m14_b_spl(x)
+
+    a = ai * (x < m14_xi1) + m14_av * ((x >= m14_xi1) & (x < m14_xi3))
+    b = bi * (x < m14_xi1) + m14_bv * ((x >= m14_xi1) & (x < m14_xi3))
```

Comment:
I know this is perfectly legal and very efficient, but I find these lines hard to read, it took me a while to realise that this should be equivalent to something like
```julia
a = zero(m14_av)
b = zero(m14_bv)
if x < m14_xi1
    # Infrared
    a = 0.574 * x^1.61
    b = -0.527 * x^1.61
elseif x < m14_xi3
    # Optical
    a = m14_a_spl(x)
    b = m14_b_spl(x)
end
```
which also saves you some computations above.  An alternative would be
```julia
a = x < m14_xi1 ? ai : x < m14_xi3 ? m14_av : zero(m14_av)
b = x < m14_xi1 ? bi : x < m14_xi3 ? m14_bv : zero(m14_bv)
```

Reply:
Wow, that reads so much better, thanks! Just changed it c042984